src/main/java/ui/issuepanel/FilterPanel.java

```diff
@@ -67,6 +80,50 @@ public FilterPanel(UI ui, IModel model, PanelControl parentPanelControl, int pan
 
         filterTextField.setKeywords(all);
     };
+    
+    private Node createNameBar() {
+        nameText = new Text(panelName);
+    
+        renameButton = new Label(RENAME_PANEL);
+        renameButton.getStyleClass().add("label-button");
+        renameButton.setId(model.getDefaultRepo() + "_col" + panelIndex + "_renameButton");
+        renameButton.setOnMouseClicked(e -> {
+            TextInputDialog renameDialog = new TextInputDialog(panelName);
+            renameDialog.getEditor().setId("panelrenameinput");
+            renameDialog.setTitle("Rename " + panelName);
+            renameDialog.setHeaderText("Enter a new name for this panel.");
+            Optional<String> result = renameDialog.showAndWait();
+            mainStage.show();
```

Comment:
I would actually like to move it to ```PanelControl``` rather than leaving it in ```FilterPanel``` because I did try using event handler here before, and found a dialog will pop for each panel instead of just the one I want to rename. I could fix it only by moving it to ```PanelControl```.